File: packages/indexer/src/indexer/embedding/safe-definition.ts
import type { ExtractContext } from "../extensions";

/** Remove authored source text from embedding projections before snapshot merge. */
export function byteSafeEmbeddingDefinition<
  T extends ReturnType<ExtractContext["define"]["definition"]>,
>(input: T): T {
  const { sourceSnippet: _sourceSnippet, ...definition } = input.definition;
  return { ...input, definition } as T;
}

File: crates/primitives/src/embedding/safe_definition.rs
use serde_json::Value;

use crate::definition::{NativeDefinitionInput, static_index_definition};

/// Preserve snippet-derived identity while excluding authored text from snapshots.
pub(crate) fn byte_safe_embedding_definition(input: NativeDefinitionInput<'_>) -> Value {
    let mut definition = static_index_definition(input);
    if let Some(object) = definition.as_object_mut() {
        object.remove("sourceSnippet");
    }
    definition
}
